### ISSUE TITLE
Search: Track / Favorites - trim search input to have match

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/OsmAnd#15291]
When searching via the search function it is already possible to find favorites and tracks. However, the prioritization of the search results is a little bit strange. When you have a favorite named "foo" and your are searching for "foo" then this favorite appears right at the top. If you search for "foo " instead (with a space at the end) then the favorite _won't_ be contained in the search results. This makes searching for favorites difficult. I'm not sure if this is a bug that needs to get fixed.

Of course it is always possible to go to My Places instead and search there for favorites. However, having to search in two different places is very cumbersome. Instead it would be nice if you could always just use the normal search menu. Here, if searching for favorites or tracks it would be of great help to limit the search results to these categories. So for this I think there should be one or two additional tabs/categories (maybe simply called "My Places" instead of "Favorites & Tracks"?).

**Comment by scaidermern** [comment]
This issue persists (or reappeared) with OsmAnd 4.3.3. Searching for "something " (with a space at the end) will not list a favorite called "something" even if it is right next to you.

Please recheck.

**Comment by sonora** [comment]
True. In search, tap the exact name of a favorite: It will show in the hit list. Then tap the space key: It dissapears.

**Comment by Zirochkabila** [comment]
OsmAnd~ 4.4.0#14038mqta, released: 2022-12-22
Reproduced - still present


https://user-images.githubusercontent.com/104760013/209112180-53131577-700f-4874-acdf-b83a409463f1.mp4

**Comment by Zirochkabila** [comment]
OsmAnd~ 4.4.0#14057mqta, released: 2022-12-28
Still present



https://user-images.githubusercontent.com/104760013/209793435-29cba805-57c3-44eb-af6a-2cceb0ac18da.mp4

**Comment by sonora** [comment]
Works now for favorites, but still not for tracks, it seems.

@Zirochkabila  Well spotted!

**Comment by Zirochkabila** [comment]
OsmAnd~ 4.4.0#14068mqta, released: 2023-01-02

Doesn't work for Favorites and Tracks in My Places

https://user-images.githubusercontent.com/104760013/210205709-0468f1fd-ae9b-45ad-bbc5-5fa3d0dd908e.mp4

**Comment by Zirochkabila** [comment]
OsmAnd~ 4.4.0#14074mqta, released: 2023-01-05
Done
